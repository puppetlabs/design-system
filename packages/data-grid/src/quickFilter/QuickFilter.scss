@import '@puppet/sass-variables/_index';

.dg-quick-filter-container {
  .dg-quick-filter-filters {
    display: flex;
    color: red;

    .dg-quick-filter {
      padding-right: 8px;
      padding-bottom: 16px;
    }

    .dg-quick-filter-empty {
      font-style: italic;

      .rc-menu-list-item-focused {
        cursor: not-allowed;
      }

      .rc-menu-list-item-disabled {
        .rc-menu-list-item-content {
          color: $puppet-n700;
        }
      }
    }
  }
}
