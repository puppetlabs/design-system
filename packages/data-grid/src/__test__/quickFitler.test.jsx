import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import QuickFilter from '../quickFilter/QuickFilter';

configure({ adapter: new Adapter() });

const filters = [
  {
    fieldLabel: 'All Operating System',
    field: 'All-Operating-System',
    options: [
      {
        value: 'linux',
        icon: 'pencil',
        label: 'linux',
      },
      {
        value: 'Windows',
        icon: 'send',
        label: 'Windows',
      },
      {
        value: 'MacOS',
        label: 'MacOS',
        icon: 'link',
      },
    ],
  },
  {
    fieldLabel: 'Puppet installed',
    field: 'Puppet-installed',
    options: [
      {
        value: 'True',
        icon: 'pencil',
        label: 'True',
      },
      {
        value: 'False',
        icon: 'send',
        label: 'False',
      },
      {
        value: 'Unknown',
        label: 'Unknown',
        icon: 'link',
      },
    ],
  },
  {
    fieldLabel: 'Empty filter',
    field: 'Empty-array',
    options: [],
  },
];

const emptyFilterLabel = 'test empty filter label';

const mockfunc = jest.fn();
const wrapper = mount(
  <QuickFilter
    filters={filters}
    onFilterSelect={mockfunc}
    emptyFilterLabel={emptyFilterLabel}
  />,
);

describe('Check total number of quick filters', () => {
  test('Number of quick filters rendered', () => {
    expect(wrapper.find('div.dg-quick-filter-filter')).toHaveLength(3);
  });
});

describe('Check number of empty quick filters', () => {
  test('Number of empty quick filters rendered', () => {
    expect(wrapper.find('div.dg-quick-filter-empty')).toHaveLength(1);
  });

  test('Empty filter label text', () => {
    expect(
      wrapper
        .find('div.dg-quick-filter-empty')
        .find('span.rc-menu-list-item-content')
        .text(),
    ).toEqual('test empty filter label');
  });
});

describe('Check component', () => {
  test('Number of dropdowns rendered ', () => {
    expect(
      wrapper.find('.dg-quick-filter-filters > .dg-quick-filter'),
    ).toHaveLength(3);
  });

  test('onFilterSelect function gets called', () => {
    wrapper.find('li.rc-menu-list-item').first().simulate('click');
    expect(mockfunc).toHaveBeenCalled();
  });
});
