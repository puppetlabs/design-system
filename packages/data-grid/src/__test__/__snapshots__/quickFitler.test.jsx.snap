// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Check component matches previous HTML snapshot 1`] = `
<QuickFilter
  emptyFilterLabel="test empty filter label"
  filters={
    Array [
      Object {
        "field": "All-Operating-System",
        "fieldLabel": "All Operating System",
        "options": Array [
          Object {
            "icon": "pencil",
            "label": "linux",
            "value": "linux",
          },
          Object {
            "icon": "send",
            "label": "Windows",
            "value": "Windows",
          },
          Object {
            "icon": "link",
            "label": "MacOS",
            "value": "MacOS",
          },
        ],
      },
      Object {
        "field": "Puppet-installed",
        "fieldLabel": "Puppet installed",
        "options": Array [
          Object {
            "icon": "pencil",
            "label": "True",
            "value": "True",
          },
          Object {
            "icon": "send",
            "label": "False",
            "value": "False",
          },
          Object {
            "icon": "link",
            "label": "Unknown",
            "value": "Unknown",
          },
        ],
      },
      Object {
        "field": "Empty-array",
        "fieldLabel": "Empty filter",
        "options": Array [],
      },
    ]
  }
  onFilterSelect={[MockFunction]}
>
  <div
    className="dg-quick-filter-container"
  >
    <div
      className="dg-quick-filter-filters"
    >
      <ButtonSelect
        className="dg-quick-filter-filter dg-quick-filter"
        id="quick-filter-All-Operating-System-0"
        key="1"
        onChange={[Function]}
        options={
          Array [
            Object {
              "icon": "pencil",
              "label": "linux",
              "value": "linux",
            },
            Object {
              "icon": "send",
              "label": "Windows",
              "value": "Windows",
            },
            Object {
              "icon": "link",
              "label": "MacOS",
              "value": "MacOS",
            },
          ]
        }
        placeholder="All Operating System"
        type="tertiary"
      >
        <ButtonSelect
          anchor="bottom left"
          applyImmediately={false}
          className="dg-quick-filter-filter dg-quick-filter"
          disabled={false}
          icon={null}
          id="quick-filter-All-Operating-System-0"
          innerFocus={false}
          loading={false}
          multiple={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "icon": "pencil",
                "label": "linux",
                "value": "linux",
              },
              Object {
                "icon": "send",
                "label": "Windows",
                "value": "Windows",
              },
              Object {
                "icon": "link",
                "label": "MacOS",
                "value": "MacOS",
              },
            ]
          }
          placeholder="All Operating System"
          style={Object {}}
          tooltipAnchor="top"
          tooltipDisabled={false}
          type="tertiary"
          value={null}
          weight="bold"
          width={null}
        >
          <div
            className="rc-button-select rc-button-select-closed dg-quick-filter-filter dg-quick-filter"
            onBlur={[Function]}
            style={Object {}}
          >
            <Button
              aria-controls="quick-filter-All-Operating-System-0-menu"
              aria-expanded={false}
              aria-haspopup="true"
              as="button"
              className="rc-button-select-target"
              disabled={false}
              forwardRefAs="ref"
              icon={null}
              iconSize="medium"
              innerFocus={false}
              loading={false}
              onClick={[Function]}
              style={null}
              trailingIcon="chevron-down"
              type="tertiary"
              weight="bold"
            >
              <button
                aria-controls="quick-filter-All-Operating-System-0-menu"
                aria-expanded={false}
                aria-haspopup="true"
                aria-label="All Operating System"
                className="rc-button rc-button-tertiary rc-button-bold rc-button-trailing-icon rc-button-full rc-button-select-target"
                disabled={false}
                onClick={[Function]}
                style={null}
                type="button"
              >
                <span
                  className="rc-button-content"
                >
                  All Operating System
                </span>
                <Icon
                  className="rc-button-icon-svg"
                  size="medium"
                  style={Object {}}
                  svg={null}
                  type="chevron-down"
                  viewBox={null}
                >
                  <svg
                    className="rc-icon rc-icon-chevron-down rc-button-icon-svg"
                    height="16px"
                    style={
                      Object {
                        "height": "16px",
                        "width": "16px",
                      }
                    }
                    viewBox="0 0 16 16"
                    width="16px"
                  >
                    <path
                      d="M8,8.58578644 L10.2928932,6.29289322 C10.6834175,5.90236893 11.3165825,5.90236893 11.7071068,6.29289322 C12.0976311,6.68341751 12.0976311,7.31658249 11.7071068,7.70710678 L8.70710678,10.7071068 C8.31658249,11.0976311 7.68341751,11.0976311 7.29289322,10.7071068 L4.29289322,7.70710678 C3.90236893,7.31658249 3.90236893,6.68341751 4.29289322,6.29289322 C4.68341751,5.90236893 5.31658249,5.90236893 5.70710678,6.29289322 L8,8.58578644 Z"
                      fillRule="evenodd"
                    />
                  </svg>
                </Icon>
              </button>
            </Button>
            <OptionMenuList
              actionLabel="Apply"
              aria-labelledby="quick-filter-All-Operating-System-0"
              autocomplete={false}
              cancelLabel="Cancel"
              className=""
              focusedIndex={0}
              footer={null}
              id="quick-filter-All-Operating-System-0-menu"
              multiple={false}
              onActionClick={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onClickItem={[Function]}
              onEscape={[Function]}
              onFocusItem={[Function]}
              options={
                Array [
                  Object {
                    "icon": "pencil",
                    "label": "linux",
                    "value": "linux",
                  },
                  Object {
                    "icon": "send",
                    "label": "Windows",
                    "value": "Windows",
                  },
                  Object {
                    "icon": "link",
                    "label": "MacOS",
                    "value": "MacOS",
                  },
                ]
              }
              selected={null}
              showCancel={false}
              style={Object {}}
            >
              <div
                className="rc-menu-list rc-option-menu-list-single"
                style={Object {}}
              >
                <ul
                  aria-activedescendant="quick-filter-All-Operating-System-0-menu-linux"
                  aria-labelledby="quick-filter-All-Operating-System-0"
                  aria-multiselectable={false}
                  className="rc-menu-list-inner"
                  id="quick-filter-All-Operating-System-0-menu"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <ForwardRef
                    disabled={false}
                    focused={true}
                    icon="pencil"
                    id="quick-filter-All-Operating-System-0-menu-linux"
                    key="linux"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    selected={false}
                    svg={null}
                    type="option"
                  >
                    <li
                      aria-selected={false}
                      className="rc-menu-list-item rc-menu-list-item-focused"
                      id="quick-filter-All-Operating-System-0-menu-linux"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="option"
                    >
                      <Icon
                        className="rc-menu-list-item-icon"
                        size="medium"
                        style={Object {}}
                        svg={null}
                        type="pencil"
                        viewBox={null}
                      >
                        <svg
                          className="rc-icon rc-icon-pencil rc-menu-list-item-icon"
                          height="16px"
                          style={
                            Object {
                              "height": "16px",
                              "width": "16px",
                            }
                          }
                          viewBox="0 0 16 16"
                          width="16px"
                        >
                          <path
                            d="M11.7,0.3 C11.3,-0.1 10.7,-0.1 10.3,0.3 L0.3,10.3 C0.1,10.5 0,10.7 0,11 L0,15 C0,15.6 0.4,16 1,16 L5,16 C5.3,16 5.5,15.9 5.7,15.7 L15.7,5.7 C16.1,5.3 16.1,4.7 15.7,4.3 L11.7,0.3 Z M4.6,14 L2,14 L2,11.4 L8,5.4 L10.6,8 L4.6,14 Z M12,6.6 L9.4,4 L11,2.4 L13.6,5 L12,6.6 Z"
                            fillRule="evenodd"
                          />
                        </svg>
                      </Icon>
                      <span
                        className="rc-menu-list-item-content"
                      >
                        linux
                      </span>
                    </li>
                  </ForwardRef>
                  <ForwardRef
                    disabled={false}
                    focused={false}
                    icon="send"
                    id="quick-filter-All-Operating-System-0-menu-Windows"
                    key="Windows"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    selected={false}
                    svg={null}
                    type="option"
                  >
                    <li
                      aria-selected={false}
                      className="rc-menu-list-item"
                      id="quick-filter-All-Operating-System-0-menu-Windows"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="option"
                    >
                      <Icon
                        className="rc-menu-list-item-icon"
                        size="medium"
                        style={Object {}}
                        svg={null}
                        type="send"
                        viewBox={null}
                      >
                        <svg
                          className="rc-icon rc-icon-send rc-menu-list-item-icon"
                          height="16px"
                          style={
                            Object {
                              "height": "16px",
                              "width": "16px",
                            }
                          }
                          viewBox="0 0 16 16"
                          width="16px"
                        >
                          <path
                            d="M7.81128337,12.9555981 L5.50946779,15.6412947 C4.9050714,16.3464882 3.74956258,15.9186292 3.75018206,14.9898706 L3.75379426,9.57429205 C3.72082332,9.38437621 3.74270714,9.18525353 3.82314037,9.00278418 C3.89674186,8.8154288 4.0262993,8.65180555 4.19850626,8.53678387 L8.41350626,5.72147137 C8.87276761,5.41471829 9.49374501,5.53835068 9.80049809,5.99761203 C10.1072512,6.45687339 9.98361878,7.07785078 9.52435743,7.38460386 L6.44520115,9.44125562 L8.02562548,10.6264612 C8.13590952,10.6664419 8.24097536,10.7266474 8.33531397,10.8075017 C8.3949363,10.8586019 8.44706376,10.9151985 8.49158929,10.9759009 L11.534996,13.258239 L13.6658118,2.60517265 L3.01204092,6.86627598 L3.46318096,7.20448619 C3.9033145,7.53444507 3.99260329,8.15869441 3.66261306,8.59878612 C3.33262282,9.03887783 2.70831415,9.12815814 2.26818061,8.79819927 L0.398726887,7.39670908 C-0.223050287,6.9305758 -0.0952406306,5.96372578 0.626304886,5.675135 L14.6332009,0.0729090649 C15.3676141,-0.220828302 16.1349518,0.41739564 15.9798265,1.19294873 L13.1784473,15.1985136 C13.0347282,15.9170408 12.1903775,16.2396066 11.6041163,15.7999525 L7.81128337,12.9555981 Z M6.20782942,11.7531219 L5.75256914,11.4117092 L5.75198667,12.284987 L6.20782942,11.7531219 Z"
                            fillRule="evenodd"
                          />
                        </svg>
                      </Icon>
                      <span
                        className="rc-menu-list-item-content"
                      >
                        Windows
                      </span>
                    </li>
                  </ForwardRef>
                  <ForwardRef
                    disabled={false}
                    focused={false}
                    icon="link"
                    id="quick-filter-All-Operating-System-0-menu-MacOS"
                    key="MacOS"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    selected={false}
                    svg={null}
                    type="option"
                  >
                    <li
                      aria-selected={false}
                      className="rc-menu-list-item"
                      id="quick-filter-All-Operating-System-0-menu-MacOS"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="option"
                    >
                      <Icon
                        className="rc-menu-list-item-icon"
                        size="medium"
                        style={Object {}}
                        svg={null}
                        type="link"
                        viewBox={null}
                      >
                        <svg
                          className="rc-icon rc-icon-link rc-menu-list-item-icon"
                          height="16px"
                          style={
                            Object {
                              "height": "16px",
                              "width": "16px",
                            }
                          }
                          viewBox="0 0 16 16"
                          width="16px"
                        >
                          <path
                            d="M4.5,16 C3.3,16 2.2,15.5 1.3,14.7 C-0.5,12.9 -0.5,10.1 1.3,8.3 L2,7.6 L3.4,9 L2.7,9.7 C1.7,10.7 1.7,12.3 2.7,13.3 C3.7,14.3 5.3,14.3 6.3,13.3 L9.3,10.3 C10.3,9.3 10.3,7.7 9.3,6.7 L8.6,6 L10,4.6 L10.7,5.3 C12.5,7.1 12.5,9.9 10.7,11.7 L7.7,14.7 C6.9,15.5 5.7,16 4.5,16 Z M6,11.4 L5.3,10.7 C3.5,8.9 3.5,6.1 5.3,4.3 L8.3,1.3 C9.2,0.4 10.3,0 11.5,0 C12.7,0 13.8,0.5 14.7,1.3 C16.5,3.1 16.5,5.9 14.7,7.7 L14,8.4 L12.6,7 L13.3,6.3 C14.3,5.3 14.3,3.7 13.3,2.7 C12.3,1.7 10.7,1.7 9.7,2.7 L6.7,5.7 C5.7,6.7 5.7,8.3 6.7,9.3 L7.4,10 L6,11.4 Z"
                            fillRule="evenodd"
                          />
                        </svg>
                      </Icon>
                      <span
                        className="rc-menu-list-item-content"
                      >
                        MacOS
                      </span>
                    </li>
                  </ForwardRef>
                </ul>
              </div>
            </OptionMenuList>
          </div>
        </ButtonSelect>
      </ButtonSelect>
      <ButtonSelect
        className="dg-quick-filter-filter dg-quick-filter"
        id="quick-filter-Puppet-installed-1"
        key="2"
        onChange={[Function]}
        options={
          Array [
            Object {
              "icon": "pencil",
              "label": "True",
              "value": "True",
            },
            Object {
              "icon": "send",
              "label": "False",
              "value": "False",
            },
            Object {
              "icon": "link",
              "label": "Unknown",
              "value": "Unknown",
            },
          ]
        }
        placeholder="Puppet installed"
        type="tertiary"
      >
        <ButtonSelect
          anchor="bottom left"
          applyImmediately={false}
          className="dg-quick-filter-filter dg-quick-filter"
          disabled={false}
          icon={null}
          id="quick-filter-Puppet-installed-1"
          innerFocus={false}
          loading={false}
          multiple={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "icon": "pencil",
                "label": "True",
                "value": "True",
              },
              Object {
                "icon": "send",
                "label": "False",
                "value": "False",
              },
              Object {
                "icon": "link",
                "label": "Unknown",
                "value": "Unknown",
              },
            ]
          }
          placeholder="Puppet installed"
          style={Object {}}
          tooltipAnchor="top"
          tooltipDisabled={false}
          type="tertiary"
          value={null}
          weight="bold"
          width={null}
        >
          <div
            className="rc-button-select rc-button-select-closed dg-quick-filter-filter dg-quick-filter"
            onBlur={[Function]}
            style={Object {}}
          >
            <Button
              aria-controls="quick-filter-Puppet-installed-1-menu"
              aria-expanded={false}
              aria-haspopup="true"
              as="button"
              className="rc-button-select-target"
              disabled={false}
              forwardRefAs="ref"
              icon={null}
              iconSize="medium"
              innerFocus={false}
              loading={false}
              onClick={[Function]}
              style={null}
              trailingIcon="chevron-down"
              type="tertiary"
              weight="bold"
            >
              <button
                aria-controls="quick-filter-Puppet-installed-1-menu"
                aria-expanded={false}
                aria-haspopup="true"
                aria-label="Puppet installed"
                className="rc-button rc-button-tertiary rc-button-bold rc-button-trailing-icon rc-button-full rc-button-select-target"
                disabled={false}
                onClick={[Function]}
                style={null}
                type="button"
              >
                <span
                  className="rc-button-content"
                >
                  Puppet installed
                </span>
                <Icon
                  className="rc-button-icon-svg"
                  size="medium"
                  style={Object {}}
                  svg={null}
                  type="chevron-down"
                  viewBox={null}
                >
                  <svg
                    className="rc-icon rc-icon-chevron-down rc-button-icon-svg"
                    height="16px"
                    style={
                      Object {
                        "height": "16px",
                        "width": "16px",
                      }
                    }
                    viewBox="0 0 16 16"
                    width="16px"
                  >
                    <path
                      d="M8,8.58578644 L10.2928932,6.29289322 C10.6834175,5.90236893 11.3165825,5.90236893 11.7071068,6.29289322 C12.0976311,6.68341751 12.0976311,7.31658249 11.7071068,7.70710678 L8.70710678,10.7071068 C8.31658249,11.0976311 7.68341751,11.0976311 7.29289322,10.7071068 L4.29289322,7.70710678 C3.90236893,7.31658249 3.90236893,6.68341751 4.29289322,6.29289322 C4.68341751,5.90236893 5.31658249,5.90236893 5.70710678,6.29289322 L8,8.58578644 Z"
                      fillRule="evenodd"
                    />
                  </svg>
                </Icon>
              </button>
            </Button>
            <OptionMenuList
              actionLabel="Apply"
              aria-labelledby="quick-filter-Puppet-installed-1"
              autocomplete={false}
              cancelLabel="Cancel"
              className=""
              focusedIndex={0}
              footer={null}
              id="quick-filter-Puppet-installed-1-menu"
              multiple={false}
              onActionClick={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onClickItem={[Function]}
              onEscape={[Function]}
              onFocusItem={[Function]}
              options={
                Array [
                  Object {
                    "icon": "pencil",
                    "label": "True",
                    "value": "True",
                  },
                  Object {
                    "icon": "send",
                    "label": "False",
                    "value": "False",
                  },
                  Object {
                    "icon": "link",
                    "label": "Unknown",
                    "value": "Unknown",
                  },
                ]
              }
              selected={null}
              showCancel={false}
              style={Object {}}
            >
              <div
                className="rc-menu-list rc-option-menu-list-single"
                style={Object {}}
              >
                <ul
                  aria-activedescendant="quick-filter-Puppet-installed-1-menu-True"
                  aria-labelledby="quick-filter-Puppet-installed-1"
                  aria-multiselectable={false}
                  className="rc-menu-list-inner"
                  id="quick-filter-Puppet-installed-1-menu"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <ForwardRef
                    disabled={false}
                    focused={true}
                    icon="pencil"
                    id="quick-filter-Puppet-installed-1-menu-True"
                    key="True"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    selected={false}
                    svg={null}
                    type="option"
                  >
                    <li
                      aria-selected={false}
                      className="rc-menu-list-item rc-menu-list-item-focused"
                      id="quick-filter-Puppet-installed-1-menu-True"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="option"
                    >
                      <Icon
                        className="rc-menu-list-item-icon"
                        size="medium"
                        style={Object {}}
                        svg={null}
                        type="pencil"
                        viewBox={null}
                      >
                        <svg
                          className="rc-icon rc-icon-pencil rc-menu-list-item-icon"
                          height="16px"
                          style={
                            Object {
                              "height": "16px",
                              "width": "16px",
                            }
                          }
                          viewBox="0 0 16 16"
                          width="16px"
                        >
                          <path
                            d="M11.7,0.3 C11.3,-0.1 10.7,-0.1 10.3,0.3 L0.3,10.3 C0.1,10.5 0,10.7 0,11 L0,15 C0,15.6 0.4,16 1,16 L5,16 C5.3,16 5.5,15.9 5.7,15.7 L15.7,5.7 C16.1,5.3 16.1,4.7 15.7,4.3 L11.7,0.3 Z M4.6,14 L2,14 L2,11.4 L8,5.4 L10.6,8 L4.6,14 Z M12,6.6 L9.4,4 L11,2.4 L13.6,5 L12,6.6 Z"
                            fillRule="evenodd"
                          />
                        </svg>
                      </Icon>
                      <span
                        className="rc-menu-list-item-content"
                      >
                        True
                      </span>
                    </li>
                  </ForwardRef>
                  <ForwardRef
                    disabled={false}
                    focused={false}
                    icon="send"
                    id="quick-filter-Puppet-installed-1-menu-False"
                    key="False"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    selected={false}
                    svg={null}
                    type="option"
                  >
                    <li
                      aria-selected={false}
                      className="rc-menu-list-item"
                      id="quick-filter-Puppet-installed-1-menu-False"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="option"
                    >
                      <Icon
                        className="rc-menu-list-item-icon"
                        size="medium"
                        style={Object {}}
                        svg={null}
                        type="send"
                        viewBox={null}
                      >
                        <svg
                          className="rc-icon rc-icon-send rc-menu-list-item-icon"
                          height="16px"
                          style={
                            Object {
                              "height": "16px",
                              "width": "16px",
                            }
                          }
                          viewBox="0 0 16 16"
                          width="16px"
                        >
                          <path
                            d="M7.81128337,12.9555981 L5.50946779,15.6412947 C4.9050714,16.3464882 3.74956258,15.9186292 3.75018206,14.9898706 L3.75379426,9.57429205 C3.72082332,9.38437621 3.74270714,9.18525353 3.82314037,9.00278418 C3.89674186,8.8154288 4.0262993,8.65180555 4.19850626,8.53678387 L8.41350626,5.72147137 C8.87276761,5.41471829 9.49374501,5.53835068 9.80049809,5.99761203 C10.1072512,6.45687339 9.98361878,7.07785078 9.52435743,7.38460386 L6.44520115,9.44125562 L8.02562548,10.6264612 C8.13590952,10.6664419 8.24097536,10.7266474 8.33531397,10.8075017 C8.3949363,10.8586019 8.44706376,10.9151985 8.49158929,10.9759009 L11.534996,13.258239 L13.6658118,2.60517265 L3.01204092,6.86627598 L3.46318096,7.20448619 C3.9033145,7.53444507 3.99260329,8.15869441 3.66261306,8.59878612 C3.33262282,9.03887783 2.70831415,9.12815814 2.26818061,8.79819927 L0.398726887,7.39670908 C-0.223050287,6.9305758 -0.0952406306,5.96372578 0.626304886,5.675135 L14.6332009,0.0729090649 C15.3676141,-0.220828302 16.1349518,0.41739564 15.9798265,1.19294873 L13.1784473,15.1985136 C13.0347282,15.9170408 12.1903775,16.2396066 11.6041163,15.7999525 L7.81128337,12.9555981 Z M6.20782942,11.7531219 L5.75256914,11.4117092 L5.75198667,12.284987 L6.20782942,11.7531219 Z"
                            fillRule="evenodd"
                          />
                        </svg>
                      </Icon>
                      <span
                        className="rc-menu-list-item-content"
                      >
                        False
                      </span>
                    </li>
                  </ForwardRef>
                  <ForwardRef
                    disabled={false}
                    focused={false}
                    icon="link"
                    id="quick-filter-Puppet-installed-1-menu-Unknown"
                    key="Unknown"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    selected={false}
                    svg={null}
                    type="option"
                  >
                    <li
                      aria-selected={false}
                      className="rc-menu-list-item"
                      id="quick-filter-Puppet-installed-1-menu-Unknown"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="option"
                    >
                      <Icon
                        className="rc-menu-list-item-icon"
                        size="medium"
                        style={Object {}}
                        svg={null}
                        type="link"
                        viewBox={null}
                      >
                        <svg
                          className="rc-icon rc-icon-link rc-menu-list-item-icon"
                          height="16px"
                          style={
                            Object {
                              "height": "16px",
                              "width": "16px",
                            }
                          }
                          viewBox="0 0 16 16"
                          width="16px"
                        >
                          <path
                            d="M4.5,16 C3.3,16 2.2,15.5 1.3,14.7 C-0.5,12.9 -0.5,10.1 1.3,8.3 L2,7.6 L3.4,9 L2.7,9.7 C1.7,10.7 1.7,12.3 2.7,13.3 C3.7,14.3 5.3,14.3 6.3,13.3 L9.3,10.3 C10.3,9.3 10.3,7.7 9.3,6.7 L8.6,6 L10,4.6 L10.7,5.3 C12.5,7.1 12.5,9.9 10.7,11.7 L7.7,14.7 C6.9,15.5 5.7,16 4.5,16 Z M6,11.4 L5.3,10.7 C3.5,8.9 3.5,6.1 5.3,4.3 L8.3,1.3 C9.2,0.4 10.3,0 11.5,0 C12.7,0 13.8,0.5 14.7,1.3 C16.5,3.1 16.5,5.9 14.7,7.7 L14,8.4 L12.6,7 L13.3,6.3 C14.3,5.3 14.3,3.7 13.3,2.7 C12.3,1.7 10.7,1.7 9.7,2.7 L6.7,5.7 C5.7,6.7 5.7,8.3 6.7,9.3 L7.4,10 L6,11.4 Z"
                            fillRule="evenodd"
                          />
                        </svg>
                      </Icon>
                      <span
                        className="rc-menu-list-item-content"
                      >
                        Unknown
                      </span>
                    </li>
                  </ForwardRef>
                </ul>
              </div>
            </OptionMenuList>
          </div>
        </ButtonSelect>
      </ButtonSelect>
      <ButtonSelect
        className="dg-quick-filter-filter dg-quick-filter dg-quick-filter-empty"
        id="quick-filter-Empty-array-2"
        key="3"
        onChange={[Function]}
        options={
          Array [
            Object {
              "disabled": true,
              "label": "test empty filter label",
              "value": "",
            },
          ]
        }
        placeholder="Empty filter"
        type="tertiary"
      >
        <ButtonSelect
          anchor="bottom left"
          applyImmediately={false}
          className="dg-quick-filter-filter dg-quick-filter dg-quick-filter-empty"
          disabled={false}
          icon={null}
          id="quick-filter-Empty-array-2"
          innerFocus={false}
          loading={false}
          multiple={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "disabled": true,
                "label": "test empty filter label",
                "value": "",
              },
            ]
          }
          placeholder="Empty filter"
          style={Object {}}
          tooltipAnchor="top"
          tooltipDisabled={false}
          type="tertiary"
          value={null}
          weight="bold"
          width={null}
        >
          <div
            className="rc-button-select rc-button-select-closed dg-quick-filter-filter dg-quick-filter dg-quick-filter-empty"
            onBlur={[Function]}
            style={Object {}}
          >
            <Button
              aria-controls="quick-filter-Empty-array-2-menu"
              aria-expanded={false}
              aria-haspopup="true"
              as="button"
              className="rc-button-select-target"
              disabled={false}
              forwardRefAs="ref"
              icon={null}
              iconSize="medium"
              innerFocus={false}
              loading={false}
              onClick={[Function]}
              style={null}
              trailingIcon="chevron-down"
              type="tertiary"
              weight="bold"
            >
              <button
                aria-controls="quick-filter-Empty-array-2-menu"
                aria-expanded={false}
                aria-haspopup="true"
                aria-label="Empty filter"
                className="rc-button rc-button-tertiary rc-button-bold rc-button-trailing-icon rc-button-full rc-button-select-target"
                disabled={false}
                onClick={[Function]}
                style={null}
                type="button"
              >
                <span
                  className="rc-button-content"
                >
                  Empty filter
                </span>
                <Icon
                  className="rc-button-icon-svg"
                  size="medium"
                  style={Object {}}
                  svg={null}
                  type="chevron-down"
                  viewBox={null}
                >
                  <svg
                    className="rc-icon rc-icon-chevron-down rc-button-icon-svg"
                    height="16px"
                    style={
                      Object {
                        "height": "16px",
                        "width": "16px",
                      }
                    }
                    viewBox="0 0 16 16"
                    width="16px"
                  >
                    <path
                      d="M8,8.58578644 L10.2928932,6.29289322 C10.6834175,5.90236893 11.3165825,5.90236893 11.7071068,6.29289322 C12.0976311,6.68341751 12.0976311,7.31658249 11.7071068,7.70710678 L8.70710678,10.7071068 C8.31658249,11.0976311 7.68341751,11.0976311 7.29289322,10.7071068 L4.29289322,7.70710678 C3.90236893,7.31658249 3.90236893,6.68341751 4.29289322,6.29289322 C4.68341751,5.90236893 5.31658249,5.90236893 5.70710678,6.29289322 L8,8.58578644 Z"
                      fillRule="evenodd"
                    />
                  </svg>
                </Icon>
              </button>
            </Button>
            <OptionMenuList
              actionLabel="Apply"
              aria-labelledby="quick-filter-Empty-array-2"
              autocomplete={false}
              cancelLabel="Cancel"
              className=""
              focusedIndex={0}
              footer={null}
              id="quick-filter-Empty-array-2-menu"
              multiple={false}
              onActionClick={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onClickItem={[Function]}
              onEscape={[Function]}
              onFocusItem={[Function]}
              options={
                Array [
                  Object {
                    "disabled": true,
                    "label": "test empty filter label",
                    "value": "",
                  },
                ]
              }
              selected={null}
              showCancel={false}
              style={Object {}}
            >
              <div
                className="rc-menu-list rc-option-menu-list-single"
                style={Object {}}
              >
                <ul
                  aria-activedescendant="quick-filter-Empty-array-2-menu-"
                  aria-labelledby="quick-filter-Empty-array-2"
                  aria-multiselectable={false}
                  className="rc-menu-list-inner"
                  id="quick-filter-Empty-array-2-menu"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="listbox"
                  tabIndex={0}
                >
                  <ForwardRef
                    disabled={true}
                    focused={true}
                    icon={null}
                    id="quick-filter-Empty-array-2-menu-"
                    key=""
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    selected={false}
                    svg={null}
                    type="option"
                  >
                    <li
                      aria-selected={false}
                      className="rc-menu-list-item rc-menu-list-item-disabled rc-menu-list-item-focused"
                      id="quick-filter-Empty-array-2-menu-"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="option"
                    >
                      <span
                        className="rc-menu-list-item-content"
                      >
                        test empty filter label
                      </span>
                    </li>
                  </ForwardRef>
                </ul>
              </div>
            </OptionMenuList>
          </div>
        </ButtonSelect>
      </ButtonSelect>
    </div>
  </div>
</QuickFilter>
`;
